drivers/common/virtio_mi/lm.c

```diff
@@ -1155,6 +1163,26 @@ virtio_vdpa_mi_dev_remove(struct rte_pci_device *pci_dev)
 	return 0;
 }
 
+static void
+virtio_ha_pf_drv_ctx_set(const struct virtio_dev_name *pf, const void *ctx)
+{
+	const struct virtio_pf_ctx *pf_ctx = (const struct virtio_pf_ctx *)ctx;
+
+	memcpy(&cached_ctx.pf_name, pf, sizeof(struct virtio_dev_name));
+	cached_ctx.vfio_group_fd = pf_ctx->vfio_group_fd;
```

Comment:
how about use dev args to pass parameter? 
like in rte_eal_hotplug_add("pci", pf_name, "vdpa=2");

Reply:
yes it's an option and was discussed before. Parav does not want to add more devargs. I am fine with either option.